### DISCUSSION TITLE
Manage Kit Environments

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#535] [Docs]
<meta path="kit/kenv-manage">
      
# Manage Kit Environments

A "kenv" (Kit Environment) is a directory with a `scripts` directory. This is the place to create a kenv to manage scripts for your dev team or projects that expose APIs (GitHub, Vercel, etc). We'll be releasing official Script Kit kenvs in the future to show off some of the best practices. 👍

Clone, create new, link, push, pull, remove. This menu allows you to manage any of the kenvs you've added to your machine.

**Comment by cfritze64** [comment]
would love more explanation of this in time. I'm nervous about messing with my GitHub setup, though it's no doubt second nature to many of your users. Mostly, I want to understand better how to set up a secondary kenv and get it syncing GitHub without disturbing syncing of my main kenv - can that be done entirely within script kit's menus, or does it require separate interaction with GitHub to establish the repo  as master? These days I'm editing scripts in Atom, and using its GitHub integration to push/pull changes between machines

**Reply by johnlindquist** [reply]
Yes, Kenvs can be controlled/versioned by the Kit menu and operate completely separate from the main scripts dir.

I'll work on more docs demos for them next week. I'm wrapping up a simple "Settings" menu today. Then finally recording lessons next week.

**Reply by ApolloTang** [reply]
hope you don't mind I ask, where are your recording lessons being published?

**Reply by johnlindquist** [reply]
The first batch of lessons has been posted here: https://egghead.io/courses/script-kit-showcase-for-optimizing-your-everyday-workflows-e20ceab4

Many, many more are in the works. scriptkit.com will eventually be updated to host courses

**Reply by ApolloTang** [reply]
that is Great. I will watch the first batch !

**Reply by joelgrimberg** [reply]
When I create a new kenv with name 'rfh' , a new kenv folder will be created, but with the name 'new'.
re-creating 'rfh' will not work ('already exists'). moving a script to a new kenv will show 'new', and not 'rfh'.  is this a bug?

**Reply by johnlindquist** [reply]
Yeah, I'm fixing the kenv scripts in the build I'm pushing out this weekend. Sorry about that
